### DISCUSSION TITLE
Docker Postgres with IPv6 UnknownHostException

### DISCUSSION THREAD

**Discussion by jkupcho** [keycloak/keycloak#15752] [Q&A]
I'm attempting to create a Docker container that will connect to a Postgres Database listening on IPv6. Working with a downloaded zip of Keycloak, the application is able to make a connection if I export JAVA_OPTS as follows:

```
$ export JAVA_OPTS=-Djava.net.preferIPv6Addresses=true
```
If I attempt to do the same within Docker, however, I receive `java.net.UnknownHostException`. I've tried both the hostname which should resolve, as well as the IP that is behind the hostname.

If I try to use the IP within the Docker container, I receive:

```
Caused by: java.net.SocketException: Protocol family unavailable
```

I'm not sure how to proceed from here, thanks!

**Comment by pravsjv** [comment]
Please refer to "Configure Keycloak Server with IPv6 or IPv4" section of the following doc for configuring IPV6 - https://www.keycloak.org/server/configuration-production. 
Hope this helps!
